kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -182,8 +230,15 @@ class KinesisSink private (
     ()
   }
 
+  case class BatchEntry(msg: ByteBuffer, key: String)
+  case class BatchEntryWithId(
+    msg: ByteBuffer,
+    kwy: String,
```

Comment:
I don't understand what do you mean? It's basically the case class instead of repeating the types.

Reply:
Exactly. My point is that `EventWithId` should just be `Event` with `Id` so that if you update the fields of `Event` it needs to happen only in one place